### ISSUE TITLE
Preloading of CSS slides background images

### ISSUE THREAD

**Issue by grks** [webslides/WebSlides#76]
Hi,

Thanks for a nice framework. Makes it a lot more fun to make presentations ;)

Just a feature suggestion for improvements. It would be nice to have some way to preload the background images. If you use large images in chrome, it doesn't look so nice even with a quick disk.

Is this something you have discussed? Maybe it would be possible to do it in a plugin? Some JS that query the document, find all background images and preloads with this trick? 

http://stackoverflow.com/questions/1373142/preloading-css-images/14390213#14390213

What do you think?

**Comment by Belelros** [comment]
Hey @grks as I at #8 there's nothing to preload. All images are downloaded whenever the page gets parsed by the browser so there aren't any tricks to do on our side.

A different scenario would be the case in which images wouldn't load *until* the slide is visible, but that's not the case (on purpose) so we can't really do anything to improve the situation you may be experiencing.

If there's anything I'm missing or something we're not seeing let me know and I'll take a look at it!

Thanks for your time!

**Comment by grks** [comment]
Ok, I looked a bit more and I think it might be related to background images loaded from file:// urls. It takes a little while to see them loading and looking through the network tab in the web inspector, it seems like the images appear loading after the slide is loaded, causing a little glitch that doesn't look too nice.

I just noticed this, loading the slide for the first time in a newly started browser. Second time slide is shown it's loaded in memory and it looks smooth. Tested in latest chromium with a 4K background.

**Comment by Belelros** [comment]
What browser are you on?

**Comment by grks** [comment]
Chromium 53.0.2785.143 Built on Ubuntu , running on Ubuntu 14.04

Actually I can reproduce it by just staying on a slide with a large background image and press ctrl-r in the browser to reload. The slide appears a fraction before the background gets loaded.

**Comment by Belelros** [comment]
I've checked on Chrome and all the backgrounds images on the media demo and all the images in general, load upon page loading. 

What you mention is normal, images are loading and that's it, nothing we can do to mitigate it. We could add a class when all the assets have finished loading but I think it's an edge case. When you're going to present you're likely going to have all in cache already and the presentation loaded.

What do you think?

**Comment by grks** [comment]
Yea, I think it's not a big issue and as you say I can just move thru the whole presentation before starting and it should be in the cache. 

It could be something with my setup, but I see a large image paint from top to bottom while it gets loaded. Might also be possible to make it look nice with some controlled animated appearing. But as you say, it's a bit of an edge case and can be circumvented easily.

**Comment by nadeemelahi** [comment]
can't be done. Well not by setting the $element.style.backgroundImage property since it takes a string.
